### PR TITLE
As/port sbh rpc

### PR DESCRIPTION
**Pull request by alsala**
backport of rpc cmds introduced for sbh wallet from zend_oo repo

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1592,6 +1705,158 @@ UniValue listtransactions(const UniValue& params, bool fHelp)
...
+
+UniValue listtxesbyaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() == 0 || params.size() > 4)
+        throw runtime_error(
+            "listtxesbyaddress ( \"address\" count from reverse_order)\n"
+            "\nReturns up to 'count' most recent transactions involving address 'address' bot for vin and vout.\n"
```
Comment (cronicc) [inline src/wallet/rpcwallet.cpp]:
```suggestion
            "\nReturns up to 'count' most recent transactions involving address 'address' both for vin and vout.\n"
```

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -1592,6 +1705,158 @@ UniValue listtransactions(const UniValue& params, bool fHelp)
...
+            "\nArguments:\n"
+            "1. \"address\"     (string, mandatory) Include transactions involving this address\n"
+            "2. count          (numeric, optional, default=10) The number of transactions to return\n"
+            "3. from           (numeric, optional, default=0) The number of transactions to skip\n"
+            "4. reverse_order  (bool, optional, default=true) sort from the most recent to the oldest\n"
+            "\nResult:\n"
+            "[\n"
+            "  {\n"
+            "      TODO\n"
```
Comment (cronicc) [inline src/wallet/rpcwallet.cpp]: Either include an actual example or remove the Result section for now. We at least should mention what the expected output is, txids or decoded transactions?

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -1592,6 +1705,158 @@ UniValue listtransactions(const UniValue& params, bool fHelp)
...
+            "\nArguments:\n"
+            " \"address\"            (string, mandatory) consider transactions involving this address\n"
+            " spendzeroconfchange  (boolean, optional) If provided the command will force zero confirmation change\n"
+            "                         spendability as specified, otherwise the value set by zend option \'spendzeroconfchange\' \n"
+            "                         will be used instead\n"
+            " includeNonFinalTxes  (boolean, optional, default=true) If true the command will consider also non final txes in the\n"
+            "                         computation of unconfirmed quantities\n"
+
+            "\nExamples:\n"
+            + HelpExampleCli("getunconfirmedtxdata", "\"ztZ5M1P9ucj3P5JaW5xtY2hWTkp6JsToiHP\"")
```
Comment (cronicc) [inline src/wallet/rpcwallet.cpp:1734]: An output example would be good.

**Review by cronicc — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (alsala) [reply src/wallet/rpcwallet.cpp]: done

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (alsala) [reply src/wallet/rpcwallet.cpp]: Ok, section removed and added a line mentioning we get decoded txes

File: src/wallet/rpcwallet.cpp (hunk 3 above)
Comment (alsala) [reply src/wallet/rpcwallet.cpp:1734]: ok done

**Review by cronicc — APPROVED** [review]

**Comment by ptagl** [comment]
These changes have already been merged with Zendoo.

**Pull request closed without merging**